src/gui/modals/FileDialog.cpp

```diff
@@ -87,22 +57,29 @@ FileDialog::FileDialog( QWidget *parent, const QString &caption,
 #endif
 
 #ifdef LMMS_BUILD_LINUX
+	QList<QUrl> m_addedSidebarUrls;
 
 	// FileSystem types : https://www.javatpoint.com/linux-file-system
-	QStringList usableFileSystems = {"ext", "ext2", "ext3", "ext4", "jfs", "reiserfs", "ntfs3", "fuse.sshfs", "fuseblk"};
+	QStringList usableFileSystems = {"ext", "ext2", "ext3", "ext4", "jfs", "reiserfs", "ntfs3", "fuse.sshfs", "fuseblk", "vfat"};
 
 	for(QStorageInfo storage : QStorageInfo::mountedVolumes())
 	{
-		storage.refresh();
-
 		if (usableFileSystems.contains(QString(storage.fileSystemType()), Qt::CaseInsensitive) && storage.isValid() && storage.isReady())
-		{			
-			urls << QUrl::fromLocalFile(storage.rootPath());	
+		{
+			auto url = QUrl::fromLocalFile(storage.rootPath());
+			if (!urls.contains(url))
+			{
+				m_addedSidebarUrls << url;
+			}
 		}
 	}
 #endif
 
+#ifdef LMMS_BUILD_APPLE
```

Comment:
No need to give apple special treatment here, use the same approach so the bug gets fixed on all platforms